x/cork/types/proposal.go

```diff
@@ -78,3 +83,37 @@ func (m *RemoveManagedCellarIDsProposal) ValidateBasic() error {
 
 	return nil
 }
+
+func NewScheduledCorkProposal(title string, description string, blockHeight uint64, targetContractAddress string, contractCallProtoJSON string) *ScheduledCorkProposal {
+	return &ScheduledCorkProposal{
+		Title:                 title,
+		Description:           description,
+		BlockHeight:           blockHeight,
+		TargetContractAddress: targetContractAddress,
+		ContractCallProtoJson: contractCallProtoJSON,
+	}
+}
+
+func (m *ScheduledCorkProposal) ProposalRoute() string {
+	return RouterKey
+}
+
+func (m *ScheduledCorkProposal) ProposalType() string {
+	return ProposalTypeScheduledCork
+}
+
+func (m *ScheduledCorkProposal) ValidateBasic() error {
+	if err := govtypes.ValidateAbstract(m); err != nil {
+		return err
+	}
+
+	if len(m.ContractCallProtoJson) == 0 {
```

Comment:
Consider if we should add further validations here -- is the JSON formatted correctly? Does `HexToAddress` below return any type of error we're ignoring or does it resolve to zero if the input isn't a valid address?